tests/network/libs/nodenetworkconfigurationpolicy.py

```diff
@@ -126,15 +137,48 @@ def wait_for_status_success(self) -> dict[str, Any]:
 
         for condition in conditions:
             if condition["type"] == Nncp.Condition.AVAILABLE:
-                self.logger.info(f"{self.kind}/{self.name} configured successfully")
-                return condition
+                if (
+                    not initial_transition_datetime
+                    or datetime.strptime(condition["lastTransitionTime"], date_format) > initial_transition_datetime
+                ):
+                    self.logger.info(f"{self.kind}/{self.name} configured successfully")
+                    return condition
             if condition["type"] == Nncp.Condition.DEGRADED:
                 raise NNCPConfigurationFailed(f"{self.name} failed on condition:\n{condition}")
         return {}
 
-    def _delete_interfaces(self) -> None:
-        desired_state = deepcopy(self.desired_state)
-        for iface in desired_state["interfaces"]:
+    def _delete_interfaces(self) -> list[Interface]:
```

Comment:
can you please rename to something like `get_interfaces_for_deletion` (as the function itself does not delete the interfaces)

Reply:
Right, now it is not deleting.
Are you ok with `interfaces_for_deletion`? I do not like `get`, it reminds me of other languages.

Reply:
Done, let me know if it is ok now.